core/src/main/scala/com/acervera/osm4scala/model/RelationEntity.scala

```diff
@@ -25,35 +25,93 @@
 
 package com.acervera.osm4scala.model
 
-import org.openstreetmap.osmosis.osmbinary.osmformat.{Relation, StringTable}
+import com.acervera.osm4scala.utilities.DecompressUtils
+import org.openstreetmap.osmosis.osmbinary.osmformat.{Info, Relation, StringTable}
 
 /**
   * Entity that represent a OSM relation as https://wiki.openstreetmap.org/wiki/Elements#Relation and https://wiki.openstreetmap.org/wiki/Relation describe
   */
-case class RelationEntity(id: Long, relations: Seq[RelationMemberEntity], tags: Map[String, String]) extends OSMEntity {
+case class RelationEntity(id: Long,
+                          relations: Seq[RelationMemberEntity],
+                          tags: Map[String, String],
+                          version: Option[Int],
+                          timestamp: Option[Long],
+                          changeset: Option[Long],
+                          uid: Option[Int],
+                          user: Option[String],
+                          visible: Option[Boolean]) extends OSMEntity {
 
   override val osmModel: OSMTypes.Value = OSMTypes.Relation
 
+  def apply(id: Long,
+            relations: Seq[RelationMemberEntity],
+            tags: Map[String, String]): RelationEntity = {
+    RelationEntity(id, relations, tags,
+      None, None, None, None, None, None)
+  }
+
+  override def toString: String = {
+    s"Relation id: ${id}, " +
+      s"relations: ${relations}, " +
+      s"tags: ${tags.toList}, " +
+      s"version: ${version.getOrElse("None")}," +
+      s"timestamp: ${timestamp.getOrElse("None")}, " +
+      s"changeset: ${changeset.getOrElse("None")}, " +
+      s"uid: ${uid.getOrElse("None")}, " +
+      s"user: ${user.getOrElse("None")}, " +
+      s"visible: ${visible.getOrElse("None")}\n"
+  }
+
 }
 
 object RelationEntity {
 
-  def apply(osmosisStringTable: StringTable, osmosisRelation: Relation): RelationEntity = {
+  val DEFAULT_DATE_GRANULARITY: Int = 1000
+
+  def apply(osmosisRelation: Relation, osmosisStringTable: StringTable): RelationEntity = {
+    apply(osmosisRelation, osmosisStringTable, Option[Int](DEFAULT_DATE_GRANULARITY))
+  }
+
+  def apply(osmosisRelation: Relation,
+            osmosisStringTable: StringTable,
+            dateGranularity: Option[Int]): RelationEntity = {
+
+    val _dateGranularity: Int = dateGranularity.getOrElse(DEFAULT_DATE_GRANULARITY)
 
     // Calculate tags using the StringTable.
-    val tags = (osmosisRelation.keys, osmosisRelation.vals).zipped.map { (k, v) =>
-      osmosisStringTable.s(k).toString("UTF-8") -> osmosisStringTable.s(v).toString("UTF-8")
+    val tags: Map[String, String] = (osmosisRelation.keys, osmosisRelation.vals).zipped.map { (k, v) =>
```

Comment:
The type is not needed because the scope is internal to the function.

Reply:
Same above. Also, I do think using map here is incorrect. We should use list or array. It could be same key two values in real life pbf and we should keep the data reader not changing anything.